torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -43,27 +48,32 @@
 class FSDPCommContext:
     """This has the communication state shared across FSDP states/parameter groups."""
 
-    def lazy_init(self):
-        if not torch.cuda.is_available():
-            raise RuntimeError("FSDP requires CUDA for streams")
+    def lazy_init(self, device: torch.device):
+        if device is None:
```

Comment:
when do we hit this branch? it mismatches the type annotation of `device: torch.device`

if we do not hit it, we should get rid of it

Reply:
if FSDPParamGroup is created with device as None, device is None. There were some test where the FSDPParamGroup device passed was None. if we allow None for FSDPParamGroup- torch.device , then we need the check. do you think that we should not allow None here?

Reply:
ah makes sense! can we try to pass a proper device in those cases and disallow `None` here? sorry about this 😢